src/zarr/testing/store.py

```diff
@@ -192,6 +192,13 @@ async def test_list_dir(self, store: S) -> None:
         assert [k async for k in store.list_dir("foo")] == []
         await store.set("foo/zarr.json", Buffer.from_bytes(b"bar"))
         await store.set("foo/c/1", Buffer.from_bytes(b"\x01"))
+        await store.set("foo/c/d/1", Buffer.from_bytes(b"\x01"))
+        await store.set("foo/c/d/2", Buffer.from_bytes(b"\x01"))
+        await store.set("foo/c/d/3", Buffer.from_bytes(b"\x01"))
+
+        keys_expected = ["foo"]
```

Comment:
Ah, this is failing on the S3 tests.

```pytb
        keys_expected = ["foo"]
        keys_observed = [k async for k in store.list_dir("")]
>       assert set(keys_observed) == set(keys_expected), keys_observed
E       AssertionError: ['test/foo']
```

It's including `self.path` (the bucket name?) in listing. Based on my reading of the tests, that should *not* be included, but I'm not 100% sure. 90940a0e01366a7339bf4abae5caca3fdcb73e30 has that fix